irma/1.0.3/Dockerfile

```diff
@@ -0,0 +1,75 @@
+# FROM defines the base docker image. This command has to come first in the file
+# The 'as' keyword lets you name the folowing stage. We use `app` for the production image
+FROM --platform=linux/x86_64 ubuntu:focal as app
+
+# ARG sets environment variables during the build stage
+ARG IRMA_VER="1.0.3"
+
+# LABEL instructions tag the image with metadata that might be important to the user
+# Optional, but highly recommended
+LABEL base.image="ubuntu:xenial"
+LABEL dockerfile.version="2"
+LABEL software="IRMA"
+LABEL software.version=$IRMA_VER
+LABEL description="IRMA was designed for the robust assembly, variant calling, and phasing of highly variable RNA viruses. Currently IRMA is deployed with modules for influenza, ebolavirus and coronavirus."
+LABEL website="https://wonder.cdc.gov/amd/flu/irma/"
+LABEL license="https://wonder.cdc.gov/amd/flu/irma/disclaimer.html"
+LABEL maintainer="John Arnn"
+LABEL maintainer.email="jarnn@utah.gov"
+
+# RUN executes code during the build
+# Install dependencies via apt-get or yum if using a centos or fedora base
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y --no-install-recommends \
+ perl \
+ default-jre \
+ apt-transport-https \
+ ca-certificates \
+ gnupg \
+ libcurl4-openssl-dev \
+ libssl-dev \
+ libxml2 \
+ libxml2-dev \
+ libncurses5 \
+ wget \
+ gzip \
+ git && \
+ apt-get autoclean && rm -rf /var/lib/apt/lists/*
+
+# IRMA Needs R
+RUN  apt-get update && apt-get -y upgrade && apt-get -y install r-base \
+  r-base-dev
+
+ # install IRMA
+RUN wget https://wonder.cdc.gov/amd/flu/irma/flu-amd-202209.zip && \
+  unzip flu-amd-202209.zip && \
+  rm flu-amd-202209.zip && \
+  chmod 755 -R flu-amd && \
+  mkdir data
+
+
+
+# set the environment
+ENV PATH="${PATH}:/flu-amd"
+
+
+# WORKDIR sets working directory
+WORKDIR /data
+
+
+FROM app as test
+
+RUN cd /flu-amd/tests && \
+  ./test_run.sh
+
+RUN wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR179/072/SRR17940172/SRR17940172_1.fastq.gz && gzip -d SRR17940172_1.fastq.gz && \
+    wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR179/072/SRR17940172/SRR17940172_2.fastq.gz && gzip -d SRR17940172_2.fastq.gz
+    
+
+#IRMA does not like the formatting for SRA paired reads; Expects Illumina Formatting; this line of code will fix this issue
+RUN awk '{if(NR%2==1) $0=sprintf($1); print;}' SRR17940172_1.fastq > SRR17940172_1a.fastq && \
+  awk '{if(NR%4==1) $0=sprintf($1" 1:N:18:NULL"); print;}' SRR17940172_1a.fastq > SRR17940172_1.fastq && \
+  awk '{if(NR%2==1) $0=sprintf($1); print;}' SRR17940172_2.fastq > SRR17940172_2a.fastq && \
+  awk '{if(NR%4==1) $0=sprintf($1" 2:N:18:NULL"); print;}' SRR17940172_2a.fastq > SRR17940172_2.fastq
```

Comment:
I'm commenting so that I'm more likely to remember the answer.

This test
1. downloads fastq files from the ENA
2. decompresses those files with gzip (these files are getting decompressed so that their header can get edited)
3. edit the header of the fastq files
4. actually runs IRMA

Reply:
Is there a formal spec for SRA paired read format?

Reply:
A formal spec from IRMA. That answer would be no and was found by troubleshooting the program. An email was sent to the IRMA developers with the reply that this was a known issue with no mention of if that would change.

Reply:
Sorry Sam I just reviewed those email messages and saw you are one of the developers/maintainers. You must have been asking about SRA format; sorry I am unaware of a formal spec but I will do a little research and ask around. See if I can turn something up.

Reply:
This was the guide I used to when writing the awk command; to get the SRA samples I wanted to test. https://www.ncbi.nlm.nih.gov/sra/docs/submitformats/#pairedend-fastq

Reply:
a quick search led me here: https://www.ncbi.nlm.nih.gov/sra/docs/sra-data-formats/

I thought I'd find more info here, since this is the home of sra-toolkit, but I don't see anything specifying the format: https://github.com/ncbi/sra-tools/wiki

Might have to contact NCBI depending on what you are looking for?

Reply:
Sorry, yes, was asking about SRA. That's good info to get started. I had a fix many months ago but no one clamored for it internally so it remained unmerged. 

I didn't also know if the format was regular or one off.

It sounds like this is causing bigger issues, so I'll try to find out from NCBI the details and merge a fix.

Thanks for the info!

Reply:
Leaving a note here to mention the SRA FASTQ compatibility issue was addressed in IRMA v1.1.2 which is now available on [Dockerhub](https://hub.docker.com/r/cdcgov/irma) and [CDC Wonder](https://wonder.cdc.gov/amd/flu/irma/). See [release notes](https://wonder.cdc.gov/amd/flu/irma/release_notes.html).

Sorry it took me so long to get this out.

Reply:
One more note: an issue fixed in v1.1.1 regressed in v1.1.2, so **v1.1.3** is now released to fix this. That version should be used instead, available at the above links.